src/main/java/org/osc/controller/nsc/utils/NSCUtils.java

```diff
@@ -0,0 +1,390 @@
+package org.osc.controller.nsc.utils;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import org.apache.log4j.Logger;
+import org.osc.controller.nsc.entities.InspectionHookNSCEntity;
+import org.osc.controller.nsc.entities.InspectionPortNSCEntity;
+import org.osc.controller.nsc.entities.MacAddressNSCEntity;
+import org.osc.controller.nsc.entities.NetworkElementNSCEntity;
+import org.osc.controller.nsc.entities.PortIpNSCEntity;
+import org.osc.sdk.controller.FailurePolicyType;
+import org.osc.sdk.controller.TagEncapsulationType;
+import org.osc.sdk.controller.element.InspectionHookElement;
+import org.osc.sdk.controller.element.InspectionPortElement;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.osgi.service.transaction.control.TransactionControl;
+
+public class NSCUtils {
+
+	private static final Logger LOGGER = Logger.getLogger(NSCUtils.class);	
+	
+	private TransactionControl txControl;
+	private EntityManager em;
+
+	public NSCUtils(EntityManager em, TransactionControl txControl) {
+		this.em = em;
+		this.txControl = txControl;
+	}
+
+	
+	public static MacAddressNSCEntity makeMacAddr(String macAddressStr) {
+    	MacAddressNSCEntity retVal = new MacAddressNSCEntity();
+    	retVal.setMacAddress(macAddressStr != null ? macAddressStr : "");
+    	return retVal;
+    }
+    
+    public static PortIpNSCEntity makePortIp(String portIpStr) {
+    	PortIpNSCEntity retVal = new PortIpNSCEntity ();
+    	retVal.setPortIp(portIpStr != null ? portIpStr : "");
+    	return retVal;
+    }
+
+    public static NetworkElementNSCEntity makeNetworkElementEntity(NetworkElement ne) {
```

Comment:
These convert methods should no longer be needed once the entity types implement the *Element interfaces.